ui/raidboss/data/07-dt/hunts/shaaloani.ts

```diff
@@ -636,7 +636,7 @@ const triggerSet: TriggerSet<Data> = {
           rotation = 3;
         }
 
-        if (pattern) {
+        if (typeof pattern === 'number') {
```

Comment:
We should early out for undefined, instead of checking for type to be number:

```ts
        if (pattern === undefined) {
          return {
            infoText: output.awayFrom!({ out: output.outOfHitbox!(), dir: awaySide }),
          };
        }

        const safeSpot = sandspoutPattern & pattern;
        let backFrontSpot = TtokSafeSpots.All;
        if (safeSpot & TtokSafeSpots.Back) {
          backFrontSpot = TtokSafeSpots.Back;
          dir1 = 'back';
        } else if (safeSpot & TtokSafeSpots.Front) {
          backFrontSpot = TtokSafeSpots.Front;
          dir1 = 'front';
        }

        if (safeSpot & backFrontSpot & TtokSafeSpots.Right) {
          dir2 = 'right';
        } else if (safeSpot & backFrontSpot & TtokSafeSpots.Left) {
          dir2 = 'left';
        }
        data.ttokSandOrbOnSet++;
        data.ttokRotated = rotation;
        return {
          alertText: output.triple!({
            inOut: output.outOfHitbox!(),
            dir2: output[dir1]!(),
            dir3: output[dir2]!(),
          }),
        };
```

Reply:
Sure, It is just that I was trying to avoid touching the logic of the code so I have only changed the condition here.